### PR TITLE
More separate code and data

### PR DESCRIPTION
**Pull request by berniev**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

Dramatic simplification of legacy code by separating code and data and applying some newer C++.

I couldn't figure out how to put the function directly under test as it is in an anonymous namespace, so kludged by copying it to the test. Better than nothing (just).

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775">pipeline status</a> for feature branch [PR_8775](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775). Pipeline [799274137](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/799274137) was triggered at [65c52b0](https://github.com/FreeCAD/FreeCAD/pull/8775/commits/65c52b092c7012aaaa5872b3fe88d38ab8fccc03). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775">pipeline status</a> for feature branch [PR_8775](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775). Pipeline [799344003](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/799344003) was triggered at [d20f7e7](https://github.com/FreeCAD/FreeCAD/pull/8775/commits/d20f7e71dee5e03c6992d80ab640a0c99fbba7d1). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
Screenshot_20230307_221952-1

@berniev please remove the whitespace so the checkbox markdown renders correctly, TIA!

**Comment by berniev** [comment]
@luzpaz Please! I did a check on the 20 or so most recent PR's. Most chop that part out completely or just ignore it. At least I went to the trouble to TRY and check the box. Obviouslsy there is something wrong with the checkbox implementation.

**Comment by luzpaz** [comment]
@berniev FTFY. please just try next time. Yes, some folk are ignoring it but I'm working on making sure people follow the protocol as best as possible.

File: src/App/Application.cpp [hunk 1]
```diff
@@ -2126,52 +2127,51 @@ void Application::initTypes()
...
-//    return make_pair(string("cmap"), string("null"));
-    else if ('@' == s[0])
-        return std::make_pair(string("response-file"), s.substr(1));
-    else
-        return make_pair(string(), string());
+
+    constexpr std::array knowns {"display",
+                                 "style",
+                                 "graphicssystem",
+                                 "widgetcount",
```
Comment (wwmayer) [inline src/App/Application.cpp:2157]: widgetcount is a special parameter and its value is an empty string, not "null".

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775">pipeline status</a> for feature branch [PR_8775](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775). Pipeline [818407778](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/818407778) was triggered at [733dcf2](https://github.com/FreeCAD/FreeCAD/pull/8775/commits/733dcf2d2efe10a02e536d9ebdf22600b98742ef). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berniev** [comment]
Thanks for spotting that. What a tricky little function it turned out to be. So much non-obvious complexity.

This function lives in an anonymous namespace with a couple of others. Is there any reason not to move these out into a separate file? Would make them testable, and shrink Application.cpp by 360 lines (from 3451).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775">pipeline status</a> for feature branch [PR_8775](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8775). Pipeline [819720337](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/819720337) was triggered at [bf8bc33](https://github.com/FreeCAD/FreeCAD/pull/8775/commits/bf8bc33d9c46778966a772a89c319e1af9d2d1a3). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**